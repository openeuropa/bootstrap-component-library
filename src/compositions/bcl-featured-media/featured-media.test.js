import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-featured-media/data/data";
import demoDataVideo from "@openeuropa/bcl-featured-media/data/data--video";
import demoDataImage from "@openeuropa/bcl-featured-media/data/data--image";
import demoFeaturedItem from "@openeuropa/bcl-featured-media/data/data--featured-item";

const template = "@oe-bcl/bcl-featured-media/featured-media.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Featured media", () => {
  test("renders correctly with iframe", () => {
    expect.assertions(1);
    return expect(render(demoData)).resolves.toMatchSnapshot();
  });

  test("renders correctly with video", () => {
    expect.assertions(1);
    return expect(render(demoDataVideo)).resolves.toMatchSnapshot();
  });

  test("renders correctly with image", () => {
    expect.assertions(1);
    return expect(render(demoDataImage)).resolves.toMatchSnapshot();
  });

  test("featured item renders correctly", () => {
    expect.assertions(1);
    return expect(render(demoFeaturedItem)).resolves.toMatchSnapshot();
  });

  test("renders correctly with iframe and with title", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoData,
        title: "Featured media test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with video and with title", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoDataVideo,
        title: "Featured media test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with image and with title", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoDataImage,
        title: "Featured media test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });

  test("featured item renders correctly with title", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoFeaturedItem,
        title: "Featured media test title",
        title_tag: "h6",
      })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with ratio", () => {
    expect.assertions(1);
    return expect(
      render({
        ...demoDataVideo,
        ratio: "21x9",
      })
    ).resolves.toMatchSnapshot();
  });
});
