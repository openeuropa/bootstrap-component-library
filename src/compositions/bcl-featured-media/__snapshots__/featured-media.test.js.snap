// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Featured media featured item renders correctly 1`] = `
<jest>
  <div
    class="bcl-featured-item pb-4"
  >
    <div
      class="row"
    >
      <h2
        class="fw-bold mb-4 bcl-heading"
      >
        A title for a featured item
      </h2>
      <div
        class="col-12 col-md-6 order-1"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
          <br />
            Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. 
          <br />
           Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
          <br />
          <br />
            Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. 
          <br />
        </p>
        <a
          href="/example.html"
        >
          Read more
        </a>
      </div>
      <div
        class="col-12 col-md-6 order-2 mt-4 mt-md-0"
      >
        <figure
          class="rounded overflow-hidden"
          src="https://www.youtube.com/embed/VFYlYeFAEH4"
        >
          <div
            class="ratio ratio-16x9"
          >
            <iframe
              allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              frameborder="0"
              height="197"
              src="https://www.youtube.com/embed/fgi-GSCB6ho"
              title="New digital strategy"
              width="350"
            />
          </div>
          <figcaption
            class="bg-lighter p-3"
          >
            Media description text goes here.
          </figcaption>
        </figure>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Featured media featured item renders correctly with title 1`] = `
<jest>
  <div
    class="bcl-featured-item pb-4"
  >
    <div
      class="row"
    >
      <h6
        class="fw-bold mb-4 bcl-heading"
      >
        Featured media test title
      </h6>
      <div
        class="col-12 col-md-6 order-1"
      >
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
          <br />
            Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. 
          <br />
           Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
          <br />
          <br />
            Duis fermentum facilisis ex, ac porta massa. Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Duis fermentum facilisis ex, ac porta massa. 
          <br />
        </p>
        <a
          href="/example.html"
        >
          Read more
        </a>
      </div>
      <div
        class="col-12 col-md-6 order-2 mt-4 mt-md-0"
      >
        <figure
          class="rounded overflow-hidden"
          src="https://www.youtube.com/embed/VFYlYeFAEH4"
        >
          <div
            class="ratio ratio-16x9"
          >
            <iframe
              allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
              allowfullscreen=""
              frameborder="0"
              height="197"
              src="https://www.youtube.com/embed/fgi-GSCB6ho"
              title="New digital strategy"
              width="350"
            />
          </div>
          <figcaption
            class="bg-lighter p-3"
          >
            Media description text goes here.
          </figcaption>
        </figure>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Featured media renders correctly with iframe 1`] = `
<jest>
  <figure
    class="rounded overflow-hidden"
    src="https://www.youtube.com/embed/VFYlYeFAEH4"
  >
    <div
      class="ratio ratio-16x9"
    >
      <iframe
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen=""
        frameborder="0"
        height="197"
        src="https://www.youtube.com/embed/fgi-GSCB6ho"
        title="New digital strategy"
        width="350"
      />
    </div>
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Featured media renders correctly with iframe and with title 1`] = `
<jest>
  <h6
    class="mb-4 bcl-heading"
  >
    Featured media test title
  </h6>
  <figure
    class="rounded overflow-hidden"
    src="https://www.youtube.com/embed/VFYlYeFAEH4"
  >
    <div
      class="ratio ratio-16x9"
    >
      <iframe
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowfullscreen=""
        frameborder="0"
        height="197"
        src="https://www.youtube.com/embed/fgi-GSCB6ho"
        title="New digital strategy"
        width="350"
      />
    </div>
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Featured media renders correctly with image 1`] = `
<jest>
  <figure
    class="rounded overflow-hidden"
  >
    <img
      alt="random image"
      class="img-fluid w-100"
      src="https://picsum.photos/1000/400?random=1"
    />
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Featured media renders correctly with image and with title 1`] = `
<jest>
  <h6
    class="mb-4 bcl-heading"
  >
    Featured media test title
  </h6>
  <figure
    class="rounded overflow-hidden"
  >
    <img
      alt="random image"
      class="img-fluid w-100"
      src="https://picsum.photos/1000/400?random=1"
    />
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Featured media renders correctly with ratio 1`] = `
<jest>
  <figure
    class="rounded overflow-hidden"
  >
    <video
      class="w-100 d-block"
      controls=""
      poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    >
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4"
        type="video/mp4"
      />
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm"
        type="video/webm"
      />
      <track
        kind="captions"
        label="English"
        src="example.vtt"
        srclang=""
      />
    </video>
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Featured media renders correctly with video 1`] = `
<jest>
  <figure
    class="rounded overflow-hidden"
  >
    <video
      class="w-100 d-block"
      controls=""
      poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    >
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4"
        type="video/mp4"
      />
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm"
        type="video/webm"
      />
      <track
        kind="captions"
        label="English"
        src="example.vtt"
        srclang=""
      />
    </video>
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;

exports[`OE - Featured media renders correctly with video and with title 1`] = `
<jest>
  <h6
    class="mb-4 bcl-heading"
  >
    Featured media test title
  </h6>
  <figure
    class="rounded overflow-hidden"
  >
    <video
      class="w-100 d-block"
      controls=""
      poster="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
    >
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.mp4"
        type="video/mp4"
      />
      <source
        src="https://inno-ecl.s3.amazonaws.com/media/videos/big_buck_bunny.webm"
        type="video/webm"
      />
      <track
        kind="captions"
        label="English"
        src="example.vtt"
        srclang=""
      />
    </video>
    <figcaption
      class="bg-lighter p-3"
    >
      Media description text goes here.
    </figcaption>
  </figure>
</jest>
`;
